runtime/executor/method.h

```diff
@@ -328,6 +328,7 @@ class Method final {
 
   size_t n_chains_;
   Chain* chains_;
+  const char* method_name_;
```

Comment:
Hi, I think `method_name_` should be in `std::string` type for safety, since the constructed `method` will be moved in the future and we also cannot control the lifecycle of `method_name`'s source.
For `BackendExecutionContext` to work, please help extend move constructor in `method.h` with `method_name_(rhs.method_name_)`.

Reply:
Hmm because we're trying control runtime size < 50kb, we can't have dependency on `std::string`...

Reply:
I have a better way to handle it now, but will need to go through internal review again. Can you use std::string to temporarily unblock the lowering work while I'm trying to get another round of review?

Reply:
I could have all work e2e in my local now. Do you want to check the cache reuse part while conducting internal proposal?